walk/config.go

```diff
@@ -46,7 +49,12 @@ func (wc *walkConfig) isExcluded(rel, base string) bool {
 	}
 	f := path.Join(rel, base)
 	for _, x := range wc.excludes {
-		if f == x {
+		matched, err := doublestar.Match(x, f)
+		if err != nil {
```

Comment:
This could get very chatty if there are a lot of files. Let's instead check for an error in `walk.Config.Configure` when we first read the pattern by calling `doublestar.Match` with a dummy file. Then we can ignore the error here.

Reply:
I've addressed this by checking the validity of the pattern during the configure phase. I've noticed also that some invalid patterns do not trigger an error with some strings but do so with other strings. I'm not sure what the best course of action here, but I ended up replacing this log with a panic so the user can file a bug with us if the checking failed.